### PR TITLE
[DSCP-81] Integrate hpack

### PR DESCRIPTION
**Pull request by volhovm**
Saved up some space :)
Also I wanted to turn on `-Werror`, but it appears that there are lots of places with warnings, so I suggest to turn it on later, after we get rid of csl-related dependencies at least.

### CODE REVIEW

File: tools/package.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+name:        disciplina-tools
+description: This package contains tools for development of Disciplina
+             platform, such as simulators of cryptographic protocols.
+
+executables:
+  disciplina-simulacrum:
+    <<: *exec-common
+    source-dirs: src/simulacrum
+    dependencies:
+      - mtl
```
Comment (Martoon-00) [inline tools/package.yaml]: Lost newline? (and +1 in module above)

Comment (volhovm) [reply tools/package.yaml]: Vim does strange things.

File: tools/disciplina-tools.cabal [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-  main-is:             Main.hs
-  build-depends:       universum
-                     , base >= 4.9
-                     , mtl
-
-  default-language:    Haskell2010
-  ghc-options:         -threaded -rtsopts
-                       -Wall
-                       -fno-warn-orphans
-                       -O2
```
Comment (Martoon-00) [inline tools/disciplina-tools.cabal]: Does `hpack` add `-O2` automatically?

Comment (volhovm) [reply tools/disciplina-tools.cabal]: It doesn't, but I doubt it was ever actually needed. Will return it though.

**Pull request merged**